+++ /var/www/vendor/bcosca/fatfree/lib/auth.php	2014-08-25 13:32:20.612715001 -0600
@@ -29,8 +29,24 @@
 		$mapper,
 		//! Storage options
 		$args;
+	
+	private $dn;
 
 	/**
+	* @return string
+	*/
+	public function getDN(){
+		return $this->dn;
+	}
+
+	/**
+	* @param string $dn
+	*/
+	public function setDN(){
+		return $this->dn;
+	}
+	
+	/**
 	*	Jig storage handler
 	*	@return bool
 	*	@param $id string
@@ -119,6 +135,7 @@
 			($info=ldap_get_entries($dc,$result)) &&
 			@ldap_bind($dc,$info[0]['dn'],$pw) &&
 			@ldap_close($dc)) {
+			$this->setDN($info[0]['dn']);
 			return $info[0]['uid'][0]==$id;
 		}
 		user_error(self::E_LDAP);
